internal/state/graph/gateway_listener.go

```diff
@@ -85,7 +83,6 @@ func newListenerConfiguratorFactory(
 			validators: []listenerValidator{
 				validateListenerHostname,
 				createAddressesValidator(gw),
-				createNoValidGatewayClassValidator(gc),
```

Comment:
I wonder if it is possible to continue to use the current approach? this way we don't need to have that special case with `buildAllInvalidListeners`, right? 

If one of the reasons is to not perform any validation if the listener is invalid, I think we can update the following loop so we can short circuit:
```
	// validators might return different conditions, so we run them all.
	for _, validator := range c.validators {
		conds = append(conds, validator(listener)...)
	}
```
Note that the conditions can be different, but in all case, there is always Accepted False condition, so we can short circuit. Maybe it's even better -- this way we don't have Accepted False and some other condition that is not related to that Accepted False.

Reply:
Spoke offline and came up with a more straightforward approach to this. We do not build the listeners if a Gateway's spec is invalid. Instead, we mark the Gateway as invalid, and when we bind the routes to the listener, we set the route attachment to false. 

If a Gateway is invalid, it will have the `Accepted/False/Invalid` condition but no Listener statuses. This is because the entire Gateway resource is invalid, so there's no reason to report the statuses of the Listeners. Any routes referencing an invalid Gateway will have the condition `Accepted/False/InvalidGateway`.